### PR TITLE
Open up some more of OffHeapHashMap

### PR DESCRIPTION
**Pull request by cschanck**
OffHeapHashMap is the base class for all the hash map functionality
int he offheap-store package, and this class traditionally has a lot
of protected members in order to expose very low-level functionality.

I opened up some more bits and bobs, in order to faciliate other
interesting usage. One slight nub was that I had to a
conditionallyShrink(), which is identical to shrink() (just a passthru)
because other classes overrode shrink().

Members (methods and fields) were only moved from privcate to protected.
Two methods were added, conditionallyShrink() and keyForEntry().

### CODE REVIEW

**Comment by terracotta-org** [comment]
Can one of the admins verify this patch?

**Comment by cschanck** [comment]
Apologize for the import deltas, I didn't realize IntelliJ has done that.

File: src/main/java/org/terracotta/offheapstore/OffHeapHashMap.java [hunk 1]
```diff
@@ -1445,6 +1447,10 @@ private void wipePendingTables() {
       }
     }
   }
+
+  protected K keyForEntry(IntBuffer entry) {
```
Comment (chrisdennis) [inline src/main/java/org/terracotta/offheapstore/OffHeapHashMap.java]: The storage engine is already protected so this new method isn't strictly required.  I'm not anti the approach here... but this feels like a one part of a changeset with a wider purpose... if we wanted to refactor some of this stuff with the aim of cleaning up then I would be in support but given everything else is just changing modifiers this feels weird.

File: src/main/java/org/terracotta/offheapstore/OffHeapHashMap.java [hunk 2]
```diff
@@ -1068,6 +1066,10 @@ protected boolean tryIncreaseReprobe() {
     }
   }
 
+  protected void conditionallyShrink() {
```
Comment (chrisdennis) [inline src/main/java/org/terracotta/offheapstore/OffHeapHashMap.java]: This is going to look weird next to the other public shrink method as this is not the conditional variant of the shrink method.  Maybe we could call this shrinkTable?

File: src/main/java/org/terracotta/offheapstore/OffHeapHashMap.java [hunk 3]
```diff
@@ -40,6 +29,15 @@
...
 
+import java.nio.ByteBuffer;
+import java.nio.IntBuffer;
+import java.util.AbstractMap;
+import java.util.AbstractSet;
+import java.util.ConcurrentModificationException;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
```
Comment (chrisdennis) [inline src/main/java/org/terracotta/offheapstore/OffHeapHashMap.java]: Could we put the import order back (as there are no actual changes to it)?

Comment (cschanck) [inline src/main/java/org/terracotta/offheapstore/OffHeapHashMap.java]: I shall try. :-)

**Pull request merged**